solr/core/src/java/org/apache/solr/cli/PostTool.java

```diff
@@ -330,15 +331,24 @@ public void runImpl(CommandLine cli) throws Exception {
       }
 
       solrUrl = SolrCLI.normalizeSolrUrl(solrUrl, true, hostContext) + hostContext;
-      String url = solrUrl + "/" + cli.getOptionValue("name") + "/update";
-      solrUpdateUrl = new URI(url);
+
+      solrUpdateUrl =
+          UriBuilder.fromUri(SolrCLI.normalizeSolrUrl(solrUrl, true, hostContext))
+              .path(hostContext)
```

Comment:
What if the `--solr-url` is a remote URL? Should we not then require the baseUrl to include the correct `hostContext`? Or have we documented a way for the user to pass in the `-DhostContext=/foo` to `bin/solr post`? I have not seen such doc, meaning hostContext will always be `/solr`, unless `bin/solr` is executed from within an installation that has overridden this in `bin/solr.in.sh`?

Reply:
The `hostContext` is indeed barely documented. The user should be able to pass a custom `hostContext` as you described via the system properties, and the `SolrCLI` class executes some custom logic to preserve these arguments as well (due to special treatment of `-D`).

I also avoided to make any changes in the bahavior and only replaced the concatenation with a URI builder.